src/tribler-gui/tribler_gui/qt_resources/mainwindow.ui

```diff
@@ -2398,6 +2398,177 @@ color: white;</string>
+                         <width>250</width>
+                         <height>28</height>
+                        </size>
+                       </property>
+                       <property name="styleSheet">
+                        <string notr="true"/>
+                       </property>
+                       <property name="placeholderText">
+                        <string>Certificate path</string>
+                       </property>
+                      </widget>
+                     </item>
+                     <item>
+                      <widget class="QToolButton" name="https_certfile_chooser_button">
+                       <property name="minimumSize">
+                        <size>
+                         <width>50</width>
+                         <height>30</height>
+                        </size>
+                       </property>
+                       <property name="baseSize">
+                        <size>
+                         <width>50</width>
+                         <height>30</height>
+                        </size>
+                       </property>
+                       <property name="toolTip">
+                        <string>Browse certificate</string>
+                       </property>
+                       <property name="text">
+                        <string>Browse</string>
+                       </property>
+                       <property name="icon">
+                        <iconset>
+                         <normaloff>../images/browse_folder.svg</normaloff>../images/browse_folder.svg</iconset>
+                       </property>
+                       <property name="toolButtonStyle">
+                        <enum>Qt::ToolButtonIconOnly</enum>
+                       </property>
+                       <property name="autoRaise">
+                        <bool>false</bool>
+                       </property>
+                       <property name="arrowType">
+                        <enum>Qt::NoArrow</enum>
+                       </property>
+                      </widget>
+                     </item>
+                    </layout>
+                   </widget>
+                  </item>
+                  <item row="19" column="1">
+                   <widget class="QLabel" name="label_65">
+                    <property name="sizePolicy">
+                     <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
+                      <horstretch>0</horstretch>
+                      <verstretch>0</verstretch>
+                     </sizepolicy>
+                    </property>
+                    <property name="text">
+                     <string>While the GUI/Core communicate over HTTP with the REST API server bound to localhost, this setting allows you to run the REST API over HTTPS, and binds to 0.0.0.0. This option can be used to remotely control Tribler. Please only use this option if you realize its security implications!</string>
```

Comment:
I guess the explanation part about the GUI/Core should be omitted, as it has nothing to do with this option. It is better to change the text to something like:


>This setting enables Tribler REST API over HTTPS, which allows you to control Tribler remotely. It binds to every available IP interface. CAUTION! Do not enable this option unless you realize the security implications!

Reply:
It's there so people know that HTTPS isn't going replace HTTP, and GUI/Core communication remains unchanged.

I know there are separate HTTP and HTTPS port settings, which already imply that it runs separately, but I just wanted to be clear.

Reply:
People looking at the GUI don't care about what kind of communication protocol is used between the Core and the GUI.
This text is here to inform the advanced users that they now got an option to enable remote access through by means of HTTPS. It is not "release notes" or Readme.txt, or changes.md, etc.

We want this kind of texts in the GUI to be as non-technical as possible.

Reply:
So, it's an option for advanced users that should be as non-technical as possible. OK, that explains it. I'll remove it 🤷

Reply:
If the GUI-oriented user can't use the feature in a few clicks, and the feature requires some technical knowledge to be used properly, and some external tool (to e.g. generate a certificate), omitting it from the GUI makes sense. Otherwise, it could cause us more trouble than good: users blaming bad usability and security.

Please, do not misunderstand me: I liked the feature initially. And I made this comment about the text to make the feature more accessible. However, your last comment, while clearly ironical, made me re-think it. As you said the feature is probably too complex for the non-technical user, especially because of the certificate thing. (Hell! Even I don't know how to get that certificate!) Our little clash just made the problem more obvious.

So, I think it is better to remove this feature from the GUI.

Reply:
I wasn't actually suggesting to remove the feature entirely from the GUI, just that bit of text concerning GUI/Core communication. Otherwise, users will never know it's even there.

Reply:
BTW, is there an easy way to generate a certificate from Windows? What's the process for setting the remote access to the Core?

Reply:
You can just use OpenSSL. I considered generating the certificate automatically with cryptography, but it's probably better to let the user handle this. Maybe the average user doesn't know how to do this, but look at it as a safety precaution: if you can't generate a certificate, you probably shouldn't be using this feature.

Reply:
Well, the kind of users who knows how to use OpenSSL directly to create a certificate needs no GUI. Or even Tribler at all...

I feel the GUI option should only be there when we start providing at least a rudimentary web interface. As it is, adding a GUI option to enable a REST interface aimed at programmers makes no sense. Unless you're enabling a remote GUI, but that's a different story..

Reply:
OK.